app/main.py

```diff
@@ -4,0 +1,52 @@
+from __future__ import annotations
+from app.data import process_json_data
+import datetime
+
+
+def shop_trip() -> None:
+    data = process_json_data()
+
+    fuel_price = data[0]
+    customers = data[1]
+    shops = data[2]
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+
+        cheapest_shop = [None, 0]
+        for shop in shops:
+            total_cost = customer.calculate_total_cost(fuel_price, shop)
+            if total_cost < cheapest_shop[1] or cheapest_shop[1] == 0:
+                cheapest_shop[0] = shop
+                cheapest_shop[1] = total_cost
+            print(f"{customer.name}'s trip to "
+                  f"the {shop.name} costs {total_cost}")
+
+        if customer.money > cheapest_shop[1]:
```

Comment:
Ensure that the condition `customer.money > cheapest_shop[1]` correctly checks if the customer has enough money for the trip and purchase. This logic assumes that `cheapest_shop[1]` is the total cost including travel and products.